interfaces/builtin/greengrass_support.go

```diff
@@ -126,31 +230,58 @@ mount options=(rw, bind) /proc/irq/ -> /proc/irq/,
 mount options=(rw, bind) /proc/sys/ -> /proc/sys/,
 mount options=(rw, bind) /proc/sysrq-trigger -> /proc/sysrq-trigger,
 
-# remap a few /proc accesses to /dev/null
+# mount some devices using /dev/null
 mount options=(rw, bind) /dev/null -> /proc/kcore,
 mount options=(rw, bind) /dev/null -> /proc/sched_debug,
 mount options=(rw, bind) /dev/null -> /proc/timer_stats,
 
-# perform the pivot_root into the overlay
-pivot_root oldroot=/var/snap/greengrass/@{SNAP_REVISION}/rootfs/.pivot_root*/ /var/snap/greengrass/*/rootfs/,
-mount options=(rw, rprivate) -> /.pivot_root*/,
-umount /.pivot_root*/,
-owner /.pivot_root*/ w,
-mount options=(rw, rprivate) -> /,
-mount options=(ro, remount, rbind) -> /,
+# umounts for tearing down containers
+umount /var/snap/{@{SNAP_NAME},@{SNAP_INSTANCE_NAME}}/*/**,
+
+# this is for container device creation
+# also need mknod and mknodat in seccomp
+capability mknod,
+
+# for the greengrassd pid file
+owner /{var/,}run/greengrassd.pid rw,
```

Comment:
This isn't listed as post pivot root. Could you use adjust the location or use a layout? (I suspect a layout might be undesirable with all the other mounts that are happening). This isn't a blocker, but wondering if it could be in SNAP_COMMON or something.

Reply:
Layouts won't work for this actually for a different reason because `/run` is explicitly disallowed by layouts, and on some systems (mine included, perhaps all but I can't say for certain) `/var/run` is a symlink to `/run` and as such can't be mounted on top of.